app/Main.hs

```diff
@@ -199,16 +199,36 @@ updateImpl config@PackageConfig{ depends } = do
   echoT ("Updating " <> pack (show (length trans)) <> " packages...")
   forConcurrently_ trans . uncurry $ installOrUpdate (set config)
 
+pursCmd :: IO Text
+pursCmd = do
+  purs <- which "purs"
+  cmd <- which "purs.cmd"
+  exe <- which "purs.exe"
+  let mpurs = msum [ constCmd "purs" purs
+                   , constCmd "purs" exe
+                   , constCmd "purs.cmd" cmd
+                   ]
+  case mpurs of
+    Nothing -> exitWithErr "The \"purs\" executable could not be found. Please make sure your PATH variable is set correctly"
+    Just c -> return c
+    where
+      constCmd :: Text -> Maybe Turtle.FilePath -> Maybe Text
+      constCmd t = fmap (const t)
```

Comment:
You can use `<$` for this.

Reply:
Ahhh, cool. Thanks! Haskell is still just a hobby for me.

I will try that out.